modules/core/04-channel/keeper/upgrade_test.go

```diff
@@ -227,103 +293,31 @@ func (suite *KeeperTestSuite) TestChanUpgradeTry() {
 
 			proofCounterpartyChannel, proofCounterpartyUpgrade, proofHeight := path.EndpointB.QueryChannelUpgradeProof()
 
-			_, err = suite.chainB.GetSimApp().IBCKeeper.ChannelKeeper.ChanUpgradeTry(
+			upgrade, err := suite.chainB.GetSimApp().IBCKeeper.ChannelKeeper.ChanUpgradeTry(
 				suite.chainB.GetContext(),
 				path.EndpointB.ChannelConfig.PortID,
 				path.EndpointB.ChannelID,
-				proposedConnectionHops,
-				upgrade.Timeout,
+				proposedUpgrade.Fields.ConnectionHops,
+				proposedUpgrade.Timeout,
 				counterpartyUpgrade,
 				path.EndpointA.GetChannel().UpgradeSequence,
 				proofCounterpartyChannel,
 				proofCounterpartyUpgrade,
 				proofHeight,
 			)
 
-			if tc.expPass {
-				suite.Require().NoError(err)
-				// suite.Require().Equal(expSequence, path.EndpointB.GetChannel().UpgradeSequence)
+			if tc.expError != nil {
```

Comment:
it would be nice to reorder this given our usual convention of 
```go
if tc.expPass {
    // no error
}
    // error
}
```

Would it be weird to do:
```go
if tc.expError == nil {
    // no error
} else {
    // error
}
```

Reply:
I think `tc.expError == nil` is fine, another option is 

```golang
expPass := tc.expError == nil

if expPass {
    // no error
} else {
    // error
}
```

Reply:
I think either suggestion is probably fine to be honest. I do prefer to have to success case assertions in the first block like our other tests :)

Reply:
I liked @chatton suggestion :) I will add `expPass := tc.expError == nil` to the top of the `suite.Run()` call near the setup, so that when you read the test you don't get too distracted by this extra assignment